model_utilities/train/finetuner.py

```diff
@@ -43,52 +44,83 @@ def __plot_losses(self, training_losses: list[float], validation_losses: list[fl
         plt.legend()
         plt.savefig("./debug/graphs/losses.png")
 
-    def __train(self, model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, dataset: ds.Dataset, optimizer: Adafactor,
+    def __tokenize_dataset(self, dataset: ds.Dataset, tokenizer: NllbTokenizerFast, max_length: int) -> Dataset:
```

Comment:
Please add a detailed, step-by-step explanation within the comments to clarify why we are tokenizing the dataset using this specific, undocumented approach. Understanding this process solely from the code is challenging without context about the data being handled.